src/main/java/games/strategy/ui/SwingComponents.java

```diff
@@ -101,8 +112,8 @@ public static ButtonGroup createButtonGroup(final JRadioButton... radioButtons)
     return group;
   }
 
-  public static JDialog newJDialog(String title) {
-    JDialog dialog = new JDialog(MainFrame.getInstance(), title);
+  public static JDialog newJDialog(final String title) {
+    final JDialog dialog = new JDialog((JFrame) null, title);
```

Comment:
Note that we're still losing the main frame reference here.

This code is currently only used to display the Unit Help and Game Notes dialogs (available from the Help menu).  I tested what happens with the actual main frame reference (I temporarily made `GameRunner#mainFrame` public), and the dialogs are still not centered relative to the window unless one adds a call to `setLocationRelativeTo()`.  However, adding a call to `setLocationRelativeTo()` in `newJDialog()` may be too early, as it shouldn't usually be done until the dialog has been `pack`ed.  Upon further inspection, both locations that use `newJDialog()` actually have calls to `setLocationRelativeTo()` that are commented out (presumably because they didn't have access to the correct frame window at the time).

So, I'm not sure it's important to preserve the main frame reference for dialogs created by `newJDialog()`.  I'm willing to leave it up to you @DanVanAtta to decide if you want to extract a `GameRunner#newJDialog()` method or simply leave it as-is.  (Actually, as modeless is the default mode for `JDialog`, another possibility is to just inline the two existing calls as `new JDialog((JFrame) null, <title>)` and remove the `newJDialog()` method completely.)